Javascript(ES5)/README.md

```diff
@@ -283,7 +356,46 @@ Forked from the excellent [Airbnb JavaScript Style Guide](https://github.com/air
     }());
     ```
 
-  - Never declare a function in a non-function block (if, while, etc). Assign the function to a variable instead. Browsers will allow you to do it, but they all interpret it differently, which is bad news bears.
+  - Use named function expressions instead of function declarations.
```

Comment:
I think the major advantage (mentioned by the guide... somewhere) is that named function expressions will give you callback information when you see a stacktrace or are debugging. Otherwise, you are stuck with a bunch of "anonymous" calls that may really hinder your debugging.

Aside from that, yes, I agree with you, it shouldn't be part of a problem as we're mostly not working on the global scope (not even in ES5), so most functions belong to a context and do have a name. Most anonymous functions are probably just small callbacks.